src/views/Home.jsx

```diff
@@ -1,7 +1,50 @@
 import './Home.css';
 import { SingleList } from '../components/SingleList';
+import { useEffect, useState, useRef } from 'react';
+import { createList } from '../api/firebase';
+import { useNavigate } from 'react-router-dom';
+
+export function Home({ data, setListPath, userEmail, userId }) {
+	const [shoppingListName, setShoppingListName] = useState('');
+	const [notificationMessage, setNotificationMessage] = useState('');
+	const navigate = useNavigate();
+	const previousDataRef = useRef(data);
+
+	useEffect(() => {
```

Comment:
Looking good. Something to notice is the the unintended navigation to list page when I'm focused on home page input box and when re-entering the home page with previous list.